transformer_engine/common/libtransformer_engine.version

```diff
@@ -1,4 +1,18 @@
 {
-	global: *nvte*; *transformer_engine*;
+	global:
+		extern "C++" {
+			nvte_*;
+			transformer_engine::cuda::sm_count*;
+			transformer_engine::cuda::sm_arch*;
+			transformer_engine::cuda::supports_multicast*;
+			transformer_engine::cuda::stream_priority_range*;
+			transformer_engine::ubuf_built_with_mpi*;
+			*transformer_engine::getenv*;
```

Comment:
We've experienced ABI problems with `getenv` before (see https://github.com/NVIDIA/TransformerEngine/issues/1283#issuecomment-2430589220 and https://github.com/NVIDIA/TransformerEngine/issues/756#issuecomment-2044108227).
```suggestion
```